### PR TITLE
Ori Changes ItemNames

### PR DESCRIPTION
**Pull request by Maxopoly**

### CODE REVIEW

**Comment by Protonull** [comment]
Might need to revert the ItemAPI updates. I think it'll mean too many plugins will need to be recompiled and staged seeming that the signatures were changed so that less exceptions are thrown, that instead booleans are returned.

**Comment by Protonull** [comment]
Just going through

 * [**RecipeAPI :: Added exit early null check**](https://github.com/CivClassic/CivModCore/pull/40/commits/f1ad79142e8935ab52ecfa424b0fb93ebc4e2b76) - Safe to merge

 * [**TextUtil :: Removed redundant replace code**](https://github.com/CivClassic/CivModCore/pull/40/commits/e2d497ce39b84b64ce653e1f3e2f12b2bdbaec99) - Safe to merge

 * [**Iteration :: Added new collect API**](https://github.com/CivClassic/CivModCore/pull/40/commits/f11f8ae76b9a04fff1f767678d5a0e30ab51feb9) - Safe to merge

 * [**NullCoalescing :: Added new cast API**](https://github.com/CivClassic/CivModCore/pull/40/commits/21753cb6c92d5210ac84e8961c3b7f764192b8ee) - Safe to merge

 * [**ResourceUtils**](https://github.com/CivClassic/CivModCore/pull/40/commits/13cf33d9d793d950eea8a28d8923b368b1da59c1) - Safe to merge

 * [**itemNames :: Refactor**](https://github.com/CivClassic/CivModCore/pull/40/commits/1c2fb4360e8c249100dcc51eaf6aaed322a35e07) - **UNSAFE TO MERGE**

 * [**ItemAPI :: Update all APIs to be less throwy**](https://github.com/CivClassic/CivModCore/pull/40/commits/b39185b4cbfea562edebf9f420decfae23a204ec) - **UNSAFE TO MERGE**

 * [**ItemAPI :: Add normalise item API**](https://github.com/CivClassic/CivModCore/pull/40/commits/9bdeaeb5e509621550f1d6f202914505d8511e8e) - Safe to merge

 * [**CustomItems :: New concept for custom items**](https://github.com/CivClassic/CivModCore/pull/40/commits/14582e253dba35dbf5037f700e239deb84d88b31) - Safe to merge

 * [**BlockAPI :: add attached face api**](https://github.com/CivClassic/CivModCore/pull/40/commits/59f26c542e544e54c5db83619481769eece65b14) - Safe to merge

 * [**BlockAPI :: rotate change**](https://github.com/CivClassic/CivModCore/pull/40/commits/fb1b2105bf526fa643a647e5e31bdeccc305e669) - Safe to merge

 * [**ItemAPI :: added getDamageable() api**](https://github.com/CivClassic/CivModCore/pull/40/commits/b1b294c9f1d5711df469b32676a9cf10322b48bc) - Safe to merge

----

"[ItemAPI :: Update all APIs to be less throwy](https://github.com/CivClassic/CivModCore/pull/40/commits/b39185b4cbfea562edebf9f420decfae23a204ec)" is unsafe by nature of it changing the signature of a few methods. The removal of annotations doesn't seem to matter to the compiler, but the change of `ItemAPI.addLore()` for example from a `void` return to a `boolean` return has non-recompiled plugins throwing fits. So whether or not to merge this is dependent on your patience to recompile and stage a lot of plugins :P I'll leave that up to you to decide.. that is what this PR is for after all ;)

"[itemNames :: Refactor](https://github.com/CivClassic/CivModCore/pull/40/commits/1c2fb4360e8c249100dcc51eaf6aaed322a35e07)" is unsafe by nature of it being a pretty substantial refactor of the item name storage and registration system. Any plugins that was previously using `LoadCustomItemNamesEvent`, which is unlikely but nonetheless possible, will break.

File: src/main/java/vg/civcraft/mc/civmodcore/api/ItemAPI.java [hunk 1]
```diff
@@ -33,10 +33,23 @@ public static boolean isValidItem(ItemStack item) {
 		if (item == null) {
 			return false;
 		}
-		if (item.getType() == Material.AIR) {
+		if (!MaterialAPI.isValidItemMaterial(item.getType())) {
```
Comment (Maxopoly) [inline src/main/java/vg/civcraft/mc/civmodcore/api/ItemAPI.java:34]: This is super unneccessary, materials that aren't items will never create ItemStacks, it throws an exception (and the material couldnt be null either)

Comment (Protonull) [reply src/main/java/vg/civcraft/mc/civmodcore/api/ItemAPI.java:34]: I don't think that's true, or at least not entirely true. I'm looking through Bukkit's code and while there might be an issue if you tried instantiating a new ItemStack with a non-item material, the error doesn't occur when setting the Material of an already created ItemStack. While I agree that the `isValidItem()` is rather over zealous with its checks, it's used in quite a few places now to absolutely ensure that an item is safe to handle.

**Pull request merged**